File: ytrue-bean/src/main/java/com/ytrue/bean/query/PageQuery.java
package com.ytrue.bean.query;

import com.baomidou.mybatisplus.core.metadata.IPage;
import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author ytrue
 * @description: PageQuery
 * @date 2022/12/21 19:06
 */
@Data
@EqualsAndHashCode(callSuper = true)
public abstract class PageQuery extends ListQuery {

    /**
     * 当前页码，默认是1
     */
    private Integer pageIndex = 1;

    /**
     * 当前页码，默认是10
     */
    private Integer pageSize = 10;


    /**
     * 组合page
     *
     * @return {@link IPage <?>}
     */
    public <T> IPage<T> page() {
        return new Page<>(pageIndex, pageSize);
    }
}

File: ytrue-admin-server-csharp/Ytrue.Models/Bases/PageQuery.cs
namespace Ytrue.Models.Bases;

public abstract class PageQuery<T> : ListQuery<T> where T : class
{
    /// <summary>
    /// 当前页码，默认是1
    /// </summary>
    public int PageIndex { get; set; } = 1;

    /// <summary>
    /// 当前页码，默认是10
    /// </summary>
    public int PageSize { get; set; } = 10;

    /// <summary>
    /// 获取Skip
    /// </summary>
    /// <returns></returns>
    public int GetSkip()
    {
        return (PageIndex - 1) * PageSize;
    }

    /// <summary>
    /// 获取Take
    /// </summary>
    /// <returns></returns>
    public int GetTake()
    {
        return PageSize;
    }
}